src/pytest_inmanta_lsm/remote_orchestrator.py

```diff
@@ -366,36 +425,68 @@ def sync_project_folder(self) -> None:
+                    # Iso4 case
+                    module = (
+                        inmanta.module.Module(self.local_project, path=str(module_path))
+                        if (module_path / inmanta.module.Module.MODULE_FILE).exists()
+                        else None
+                    )
+                else:
+                    # Iso5+ case
+                    module = inmanta.module.Module.from_path(module_path)
+
+                if module is None:
+                    LOGGER.warning("%s is not a valid module, it will be skipped", str(module_path))
+                    continue
 
-                modules.add(module.name)
+                remote_module_path = libs_path / module.name
+                self.sync_local_folder(module_path, remote_module_path, excludes=[])
 
-        libs_path = str(self.remote_project_path / "libs")
+                synced_modules.add(module.name)
 
-        if len(modules) > 0:
-            # Make sure all the modules we synced appear to be version controlled
-            mkdir_module = ["mkdir"] + [x for module in modules for x in ["-p", module + "/.git"]]
-            self.run_command(mkdir_module, cwd=libs_path)
+        # Sync all the modules in editable mode which were not found in the project module's paths
+        for module in modules.values():
+            if module.name in synced_modules:
+                # The module has already been synced
+                continue
 
-        if len(modules) > 0:
-            # Delete all modules which are on the remote libs folder but we didn't sync
-            grep_extra = ["grep", "-v"] + [x for module in modules for x in ["-e", module]]
-            grep_extra_cmd = shlex.join(grep_extra)
-            clear_extra = f"rm -rf $(ls . | {grep_extra_cmd} | xargs)"
-        else:
-            # Delete all modules
-            clear_extra = "rm -rf *"
+            if hasattr(inmanta.module, "ModuleV2") and isinstance(module, inmanta.module.ModuleV2) and not module.is_editable():
+                # Module v2 which are not editable installs should not be synced
+                continue
+
+            # V1 modules and editable installs should be synced
+            self.sync_local_folder(
+                local_folder=pathlib.Path(module._path),  # Use ._path instead of .path to stay backward compatible with iso4
+                remote_folder=libs_path / module.name,
+                excludes=[],
+            )
 
-        self.run_command([clear_extra], shell=True, cwd=libs_path)
+            synced_modules.add(module.name)
+
+        # Make sure all the modules we synced appear to be version controlled
+        mkdir_module = ["mkdir"] + [x for module in synced_modules for x in ["-p", module + "/.git"]]
+        self.run_command(mkdir_module, cwd=str(libs_path))
+
+        # Delete all modules which are on the remote libs folder but we didn't sync
+        grep_extra = ["grep", "-v"] + [x for module in synced_modules for x in ["-e", module]]
```

Comment:
To be honest I struggle to understand the alternative solution you propose.  What I want to do is a simple flatmap, and to me the most easy way to achieve this in python is via list comprehension

Reply:
`itertools.chain.from_iterable` is basically flatten, though I admit it's verbose (alas Python doesn't have a `flatmap` method). I personally find it more readable than the nested list comprehension even so, but that might be because I just replace it with flatten in my head. I find it easier to parse. But like I said, this is a very subjective thing, I'm completely fine to leave it as is if you find the list comprehension approach more readable.